hooks/expand_show_tree.py

```diff
@@ -0,0 +1,125 @@
+
+        :return: A list of paths in the order of Test_User:Shot:Seq:Project,
+        code that calls it can decide what to do with that paths.
+        """
+        # this is (:) separated list of paths
+        show_paths = self.expand_tree(bundle_obj, additional_subdirectories, path)
+        paths_list = show_paths.split(":")
+
+        # let's return the path list and let the code that calls it decide what to do
+        return paths_list
+
+        # we need to return the first path since that is the override we have configured
+        # Seq:Shot:Project
+        # if paths_list:
+        #     return paths_list[0]
+
+        # none of the paths exist! the hook was configured without actually populating the show with the hook.
+        # return ''
+
+    @staticmethod
+    def expand_tree(bundle_obj, additional_subdirectories, *paths):
+        """Build search paths to form a lookup-tree where each path
+        is appended to a standard location prefix starting from the
+        bottom, walking up to the top.
+
+        >>> self.expand_tree('$DD_OS/solibs', 'lib/xyz').split(":")
+        ['/dd/shows/ABC/RD/cent6/solibs',
+         '/dd/shows/ABC/RD/lib/xyz',
+         '/dd/shows/ABC/cent6/solibs',
+         '/dd/shows/ABC/tools/lib/xyz']
+
+        The generated platform-dependent path is returned with
+        variables expanded to values contained in this instance.
+        The returned path is generated by invoking :meth:`sgtk.dd_xplatform_utils.build_path`.
+
+        """
+        context = bundle_obj.context
+        default_root = bundle_obj.tank.pipeline_configuration.get_primary_data_root()
+
+        if context.project is None:
+            # our context is completely empty!
+            # there is no show tree to expand these paths on
+            return ''
+
+        prefixes = []
+        if context.entity is None:
+            # we have a project but not an entity
+            prefixes.extend([
+                default_root,
+            ])
+
+        if context.entity:
+            # we have an entity
+
+            if context.entity["type"] == "Sequence":
+                # add the sequence paths to the mix
+                prefixes.extend([
+                    os.path.join(default_root, context.entity["name"]),
+                    default_root,
+                ])
+
+            if context.entity["type"] == "Shot":
+                entities_by_type = dict([(x["type"], x) for x in context.additional_entities])
+                seq_entity = entities_by_type["Sequence"]
+                # add the sequence and shot paths to the mix
+                prefixes.extend([
+                    os.path.join(default_root, seq_entity["name"], context.entity["name"]),
+                    os.path.join(default_root, seq_entity["name"]),
+                    default_root,
+                ])
+
+            # let's just support project level overrides for an Asset entity
+            if context.entity["type"] == "Asset":
```

Comment:
You should be able to get `sequence` and `shot` from the additional_entities, and should use those levels.